contrib/mesos/pkg/scheduler/podtask/pod_task.go

```diff
@@ -220,6 +220,19 @@ func (t *T) AcceptOffer(offer *mesos.Offer) bool {
 		return false
 	}
 
+	// check the NodeSelector
+	labels := map[string]string{}
+	for _, a := range offer.Attributes {
+		if a.GetType() == mesos.Value_TEXT {
+			labels[*a.Name] = *a.GetText().Value
```

Comment:
kubernetes gives us sets and selectors. we should use them.
https://github.com/kubernetes/kubernetes/blob/master/plugin/pkg/scheduler/algorithm/predicates/predicates.go#L198